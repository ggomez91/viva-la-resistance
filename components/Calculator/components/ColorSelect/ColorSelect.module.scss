.ColorSelect {
  width: 42px;

  select {
    text-shadow: 0px 0px 5px #000;
    height: 70px;
    background: transparent;
  }
}
