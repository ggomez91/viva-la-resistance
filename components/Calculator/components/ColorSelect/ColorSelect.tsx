import React from "react";
import styles from "./ColorSelect.module.scss";
import colors from "../../domain/colors";

type ColorSelectProps = {
  color: string;
  setColor: (color: string) => void;
};

/**
 * Renders a colored select with the known colors
 */
const ColorSelect: React.FC<ColorSelectProps> = ({ color, setColor }) => {
  return (
    <div className={styles.ColorSelect}>
      <select
        value={color}
        title="Select your surfboard"
        className="selectpicker"
        onChange={(e) => {
          setColor(e.target.value);
        }}
        style={{
          backgroundColor:
            color == null || color == "none" ? "transparent" : color,
        }}
      >
        {colors.map((color) => (
          <option
            key={color}
            value={color}
            style={{ backgroundColor: color }}
          ></option>
        ))}
      </select>
    </div>
  );
};

export default ColorSelect;
