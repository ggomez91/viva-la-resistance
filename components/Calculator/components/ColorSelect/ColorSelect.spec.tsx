import React from "react";
import renderer from "react-test-renderer";
import ColorSelect from "./ColorSelect";

describe("ColorSelect", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ColorSelect />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
