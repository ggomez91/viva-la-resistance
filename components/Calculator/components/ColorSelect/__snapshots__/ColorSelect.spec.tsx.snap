// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelect renders correctly 1`] = `
<div>
  <select
    className="selectpicker"
    onChange={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
    title="Select your surfboard"
  >
    <option
      style={
        Object {
          "backgroundColor": "pink",
        }
      }
      value="pink"
    />
    <option
      style={
        Object {
          "backgroundColor": "silver",
        }
      }
      value="silver"
    />
    <option
      style={
        Object {
          "backgroundColor": "gold",
        }
      }
      value="gold"
    />
    <option
      style={
        Object {
          "backgroundColor": "black",
        }
      }
      value="black"
    />
    <option
      style={
        Object {
          "backgroundColor": "brown",
        }
      }
      value="brown"
    />
    <option
      style={
        Object {
          "backgroundColor": "red",
        }
      }
      value="red"
    />
    <option
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      value="orange"
    />
    <option
      style={
        Object {
          "backgroundColor": "yellow",
        }
      }
      value="yellow"
    />
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value="green"
    />
    <option
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
      value="blue"
    />
    <option
      style={
        Object {
          "backgroundColor": "violet",
        }
      }
      value="violet"
    />
    <option
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      value="gray"
    />
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value="white"
    />
  </select>
</div>
`;
